source/s3_client.c

```diff
@@ -319,9 +322,22 @@ PyObject *aws_py_s3_client_new(PyObject *self, PyObject *args) {
         signing_config = &default_signing_config;
     }
 
+    struct aws_byte_cursor *network_interface_names = NULL;
```

Comment:
ugh. the code above this creates `anonymous_credentials` that needs cleanup if error occurs.

so like ... the location of the `* From hereon, we need to clean up if errors occur */` comment isn't accurate anymore

maybe just declare these higher up, and move the rest of the code for processing `network_interface_names_py` down in one place. Instead of splitting between 1 no-need-to-cleanup section and 1 yes-you-gotta-cleanup section

Reply:
Thanks, I have tried to fix the cleanup logic and make the comment accurate.